### PR TITLE
style: fix formatting and clippy nags

### PR DESCRIPTION
**Pull request by tshauck**
also adds .github tests

before this is merged, I'll need to update the build actions so that the formatting tests aren't run every push, and more like pushes to `main` and open PRs.

### CODE REVIEW

File: src/async_reader/zarr_read_async.rs [hunk 1]
```diff
@@ -32,16 +31,16 @@ pub struct ZarrPath {
...
-        Self {store, location}
+        Self { store, location }
     }
 }
 
 /// Implementation of the [`ZarrReadAsync`] trait for a [`ZarrPath`] which contains the
 /// object store that the data will be read from.
 #[async_trait]
-impl ZarrReadAsync for ZarrPath {
+impl<'a> ZarrReadAsync<'a> for ZarrPath {
```
Comment (maximedion2) [inline src/async_reader/zarr_read_async.rs:43]: So, is adding the lifetimes and replacing vectors with slices something clippy complains about? Oh, wait adding the lifetime is _because_ the vectors were replaced with slices right? Do you know why it doesn't like vectors there, I mean sure they might have a tiny bit of overhead, but still...?

**Comment by maximedion2** [comment]
So to confirm, were you able to run the tests now (after rebasing and after I made the `test-data` repo public)?

**Comment by tshauck** [comment]
image

File: src/async_reader/zarr_read_async.rs (hunk 1 above)
Comment (tshauck) [reply src/async_reader/zarr_read_async.rs:43]: This is the error clippy shows: https://rust-lang.github.io/rust-clippy/master/index.html#/ptr_arg

As an aside, if you don't have clippy setup, it's pretty easy to use/run and obviously after this PR is merged it'll be required.

Comment (maximedion2) [reply src/async_reader/zarr_read_async.rs:43]: Yep, I'll set it up before my next merge.

**Comment by maximedion2** [comment]
> also adds .github tests
> 
> before this is merged, I'll need to update the build actions so that the formatting tests aren't run every push, and more like pushes to `main` and open PRs.

Is there still something you want to update? Looking at the file, it seems like it's still running on all pushes, I do thing only running on PRs to main would make sense?

**Comment by tshauck** [comment]
Yep, should be updated

**Comment by maximedion2** [comment]
> Yep, should be updated. I just have it do all PRs, feels like a pretty rare occurrence of a PR into a non-`main` branch anyways 🤷

Yeah that's fine, works for me, merging now!

**Review by maximedion2 — APPROVED** [review]

**Pull request merged**